src/AutoRest.CSharp/Common/Generation/Writers/RequestWriterHelpers.cs

```diff
@@ -269,8 +271,13 @@ private static void WriteHeader(CodeWriter writer, CodeWriterDeclaration request
 
             using (WriteValueNullCheck(writer, header.Value))
             {
-                writer.Append($"{request}.Headers.{method}({header.Name:L}, ");
-
+                if (header is RequestConditionsHeader conditionsHeader)
```

Comment:
You can do the type check here and avoid creating special `RequestConditionsHeader` type:
```cs
private static bool IsComplexHeaderType(CSharpType type) 
    => type.Equals(typeof(MatchConditions)) || type.Equals(typeof(RequestCondition))
```
```cs
if (IsComplexHeaderType(header.Value.Type)) 
{
    writer.Append($"{request}.Headers.{method}(");
} 
else
{
    writer.Append($"{request}.Headers.{method}({header.Name:L}, ");
}
```

You can also extract this logic into `IsKnownHeaderType(header.Value.Type)`